server/schedule/filter/filters.go

```diff
@@ -672,3 +677,29 @@ func (f *isolationFilter) Target(opt opt.Options, store *core.StoreInfo) bool {
 	}
 	return true
 }
+
+// createRegionForRuleFit is used to create a clone region with RegionCreateOptions which is only used for
+// FitRegion in filter
+func createRegionForRuleFit(startKey, endKey []byte,
+	peers []*metapb.Peer, leader *metapb.Peer, opts ...core.RegionCreateOption) *core.RegionInfo {
+	copyLeader := &metapb.Peer{
+		Id:      leader.Id,
+		StoreId: leader.StoreId,
+		Role:    leader.Role,
+	}
+	copyPeers := make([]*metapb.Peer, 0, len(peers))
```

Comment:
Seems we don't have to copy leader. And if the peer list is untouched, we don't need to copy it either.

Reply:
I think for both `ruleFitFilter` and `ruleLeaderFitFilter`, they will both touch peers and leaders. I think it is necessary to copy peers and leaders either.

Reply:
It makes sense.